File: experiments/communication/httpsms/swiftui/Activity_login.swift
import SwiftUI

struct Activity_login: View {
  var body: some View {
    VStack(alignment: .leading, spacing: 0) {
      Image(systemName: "photo").resizable().scaledToFit().frame(width: 100.0, height: 100.0).padding(.top, 32.0)
      Text("Opennhttpsms.com/settingsnto get your API key").font(.system(size: 20)).multilineTextAlignment(.center).padding(.bottom, 20.0)
      VStack(alignment: .leading, spacing: 8) {
        Text("API Key")
        TextField("", text: .constant("")).frame(maxWidth: .infinity)
      }
      .frame(maxWidth: .infinity)
      VStack(alignment: .leading, spacing: 8) {
        Text("Phone number (SIM1)")
        TextField("", text: .constant("")).frame(maxWidth: .infinity)
      }
      .frame(maxWidth: .infinity)
      .padding(.bottom, 224.0)
      VStack(alignment: .leading, spacing: 8) {
        Text("Phone number (SIM2)")
        TextField("", text: .constant("")).frame(maxWidth: .infinity)
      }
      .frame(maxWidth: .infinity)
      .padding(.bottom, 224.0)
      VStack(alignment: .leading, spacing: 0) {
        VStack(alignment: .leading, spacing: 8) {
          Text("Server URL")
          TextField("", text: .constant("")).frame(maxWidth: .infinity)
        }
        .frame(maxWidth: .infinity)
      }
      .frame(maxWidth: .infinity)
      VStack(alignment: .leading, spacing: 0) {
        VStack(alignment: .leading, spacing: 0) {
        }
        .padding(10.0)
        VStack(alignment: .leading, spacing: 0) {
        }
        .frame(maxWidth: .infinity)
        .padding(.top, 4.0)
      }
      .frame(maxWidth: .infinity)
    }
    .frame(maxWidth: .infinity, maxHeight: .infinity)
  }
}

struct Activity_login_Previews: PreviewProvider {
  static var previews: some View {
    Activity_login()
  }
}

File: experiments/communication/httpsms/compose/activity_login.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun activity_login() {
  Column(modifier = Modifier.fillMaxWidth().fillMaxHeight(), verticalArrangement = Arrangement.Top, horizontalAlignment = Alignment.Start) {
    Box(modifier = Modifier.size(100.0.dp, 100.0.dp).padding(top = 32.0.dp)) { /* TODO Image */ }
    Text(text = "Opennhttpsms.com/settingsnto get your API key", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(bottom = 20.0.dp), fontSize = 20.sp, textAlign = TextAlign.Center)
    Column(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
      Text("API Key")
      OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.fillMaxWidth().wrapContentHeight())
    }
    Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(bottom = 224.0.dp)) {
      Text("Phone number (SIM1)")
      OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.fillMaxWidth().wrapContentHeight())
    }
    Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(bottom = 224.0.dp)) {
      Text("Phone number (SIM2)")
      OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.fillMaxWidth().wrapContentHeight())
    }
    Column(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
      Column(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
        Text("Server URL")
        OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.fillMaxWidth().wrapContentHeight())
      }
    }
    Column(modifier = Modifier.wrapContentWidth().wrapContentHeight().weight(50.0f)) {
      Column(modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(10.0.dp)) {
      }
      Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(top = 4.0.dp)) {
      }
    }
  }
}


@Preview(showBackground = true)
@Composable
fun activity_login_preview() {
    MyApplicationTheme {
        activity_login()
    }
}
